### PR TITLE
Only return Saved Cards/Contracts tied to Customer Selected Currency

### PR DESCRIPTION
**Pull request by ChantalDelachevrotiere**
Contracts can only be used with the currency the contract was opened with. 

When presenting the saved cards (contracts) to the customer only saved cards (contracts) with the correct currency will be displayed.

### CODE REVIEW

File: Block/Customer/Savedcard.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function getSavedCards()
         $customerCards = $this->getCustomerCards();
         if ($customerCards && $customerCards->count()) {
             foreach ($customerCards as $card) {
-                $cards[]=['type'=>$card->getMethod(),'number'=>$card->getIdentifier(),'contract_id'=>$card->getId()];
+                $cards[]=['type'=>$card->getMethod(),'number'=>$card->getIdentifier(),'contract_id'=>$card->getId(), 'currency'=>$card->getcurrency()];
```
Comment (grandel-withreach) [inline Block/Customer/Savedcard.php]: Nit pick, but could we format this array to break each key-value pair into its own line

ex.

```
[
    'type' => $card->getMethod(),
    'number'=>$card->getIdentifier(),
    ...
]
```

File: Model/CcConfigProvider.php [hunk 2]
```diff
@@ -106,7 +109,8 @@ public function getConfig()
                 'reach_cc' => [
                     'oc_enabled'=>(boolean)$this->reachHelper->getAllowOpenContract(),
                     'open_contracts'=>$this->getOpenContracts(),
-                     //got idea from here: https://webkul.com/blog/adding-additional-variables-in-window-checkoutconfig-on-magento-2-checkout-page/
+                    'selected_currency'=>$this->_storeManager->getStore()->getCurrentCurrency()->getCode(),
```
Comment (grandel-withreach) [inline Model/CcConfigProvider.php:112]: Oh nice! you were able to determine where the currency is being stored!

File: view/frontend/web/js/view/payment/method-renderer/cc.js [hunk 3]
```diff
@@ -238,6 +240,7 @@ define([
                 {
                     return false;
                 }
+                console.log("herererere")
```
Comment (grandel-withreach) [inline view/frontend/web/js/view/payment/method-renderer/cc.js]: I presume this was unintentional 😂

File: Model/CcConfigProvider.php [hunk 4]
```diff
@@ -134,7 +138,7 @@ protected function getOpenContracts()
             $collection = $this->openContract->getCollection();
             $collection->addFieldToFilter('customer_id', ['eq'=>$customerId]);
             foreach ($collection as $contract) {
-                $contracts[] = ['contractId'=>$contract->getReachContractId(),'label'=>$contract->getMethod().' - '.$contract->getIdentifier()];
+                $contracts[] = ['contractId'=>$contract->getReachContractId(),'label'=>$contract->getMethod().' - '.$contract->getIdentifier(), 'contractCurrency'=>$contract->getCurrency()];
```
Comment (grandel-withreach) [inline Model/CcConfigProvider.php]: Same formatting comment from above

File: Model/CcConfigProvider.php (hunk 2 above)
Comment (ChantalDelachevrotiere) [reply Model/CcConfigProvider.php:112]: Well, sorta. At the time the contracts are being rendered on the FE the selected currency is not available. But this line of code solves that

File: view/frontend/web/js/view/payment/method-renderer/cc.js (hunk 3 above)
Comment (ChantalDelachevrotiere) [reply view/frontend/web/js/view/payment/method-renderer/cc.js]: 💀

**Review by grandel-withreach — APPROVED** [review]

**Pull request merged**